cedar-policy-core/src/validator/json_schema.rs

```diff
@@ -1399,6 +1483,176 @@ impl Type<ConditionalName> {
     }
 }
 
+impl Type<InternalName> {
+    /// Resolve EntityOrCommon types to specific Entity or CommonType designations
+    pub fn resolve_type_entity_or_common(
```

Comment:
nit: I find it a little messy that similar functions are named `resolve_type_entity_or_common()` here, `fully_resolve_common_and_entity_types()` on `NamespaceDef`, and `resolve_entity_or_common_types()` on `Fragment`. Is one of these names better than the others? should we use the same name everywhere?